### PR TITLE
fixes #7560 - add :required => true on host fields that have conditional validation if host.managed?

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/views/hosts/_form.html.erb [hunk 1]
```diff
@@ -34,8 +34,9 @@
             { :include_blank => !@host.managed?, :selected => @organization.try(:id)},
             { :disabled => !@host.new_record?,
               :onchange => 'organization_changed(this);', :label => _("Organization"), :'data-host-id' => @host.id,
-            :'data-url' => process_taxonomy_hosts_path,
-            :help_inline => :indicator } %>
+              :'data-url' => process_taxonomy_hosts_path,
+              :help_inline => :indicator,
+              :required => SETTINGS[:locations_enabled] } %>
```
Comment (dLobatog) [inline app/views/hosts/_form.html.erb]: This should be `SETTINGS[:organizations_enabled]`

File: app/views/hosts/_form.html.erb [hunk 2]
```diff
@@ -44,7 +45,8 @@
             { :disabled => !@host.new_record?,
               :onchange => 'location_changed(this);', :label => _("Location"), :"data-host-id" => @host.id,
               :'data-url' => process_taxonomy_hosts_path, :selected => Location.my_locations,
-              :help_inline => :indicator } %>
+              :help_inline => :indicator,
+              :required => SETTINGS[:organizations_enabled] } %>
```
Comment (dLobatog) [inline app/views/hosts/_form.html.erb]: This should be `SETTINGS[:locations_enabled]`

**Comment by dLobatog** [comment]
Some required fields are missing:
- Operating System
  - Provisioning Method
  - Media
  - Partition table
  - Root password

There's some fields under Virtual Machine also that are required, depending on which Compute Resource we're using, I guess we can leave that out for the moment unless you feel like testing for all of them.

**Comment by isratrade** [comment]
@eLobato, I added `:required => true` to the fields `provisioning_method`, `medium_id`, `ptable_id`, and `root_pass`. I checked in the UI that these fields are not visible when they are not required.

**Comment by isratrade** [comment]
[test]

**Comment by dLobatog** [comment]
@isratrade something happened in Jenkins, it marked one of the jobs as SIGKILLED. It's probably fine but [test]. I know the fields are only shown under certain circumstances but it might be a bit more consistent from the user point of view that all required fields are marked. Thanks for the update!

**Comment by dLobatog** [comment]
Merged as https://github.com/theforeman/foreman/commit/571614aec259046da6c19843518ba8e78e7f2f94 , thank you @isratrade !

**Pull request closed without merging**